### PR TITLE
[runtime hash] fix runtime hash mismatch between inside `app` image and in "development mode"

### PR DESCRIPTION
**Pull request by xingyaoww**
**Short description of the problem this fixes or functionality that this introduces. This may be used for the CHANGELOG**

As mentioned in https://github.com/All-Hands-AI/OpenHands/issues/3767, the runtime hash you got inside `app` image and you got in development mode is different. This causes issues and requires us to manually tag "runtime" images when releasing a new version -. In contrast, ideally, these two hashs should be identical, so there's no need to specify runtime images manually when release.

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

This PR 
- fixes the issue of hash mismatch by allowing `dirhash` to ignore `.pyc` (which seems to exist in the `app` docker image) and hidden folder (there's a `openhands/.cache` inside `app` image). ---- the root cause is actually `python -m build` does not respect the `MANEFEST.in` inside `app` docker, which i tweaked a bit but didn't find a good solution there.
- Add a test in GHCR workflow to tests for this to prevent future degradation

---
**Link of any specific issues this addresses**

https://github.com/All-Hands-AI/OpenHands/issues/3767

### CODE REVIEW

**Comment by tobitege** [comment]
In "Build app image": `docker: invalid reference format: repository name (All-Hands-AI/openhands) must be lowercase.`
https://github.com/All-Hands-AI/OpenHands/actions/runs/11021605267/job/30609063421?pr=4039#step:9:9

**Comment by xingyaoww** [comment]
ha! This finally works - it should be ready for review!

image

**Comment by mamoodi** [comment]
Not entirely sure I understand the hash parts or what the issue is but the new job looks fine. Everything runs in parallel so doesn't add time as well which is a plus.

**Review by neubig — APPROVED** [review]
LGTM, given that you've confirmed that it works!

File: .github/workflows/ghcr-build.yml [hunk 1]
```diff
@@ -25,7 +25,69 @@ on:
...
+          # if set to "true" but frees about 6 GB
+          tool-cache: true
+          # all of these default to true, but feel free to set to
+          # "false" if necessary for your workflow
+          android: true
+          dotnet: true
+          haskell: true
+          large-packages: true
+          docker-images: false
+          swap-storage: true
```
Comment (rbren) [inline .github/workflows/ghcr-build.yml:57]: I think we've managed to ditch this?

**Review by rbren — APPROVED** [review]

File: .github/workflows/ghcr-build.yml (hunk 1 above)
Comment (xingyaoww) [reply .github/workflows/ghcr-build.yml:57]: I think @tobitege  add it back because of `buildx` https://github.com/All-Hands-AI/OpenHands/pull/4009

Comment (tobitege) [reply .github/workflows/ghcr-build.yml:57]: Yes, I did. It saves us avoidable trouble and it is only like 2-3 minutes.

**Pull request merged**